data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -74,111 +77,106 @@ private TableMigrationUtil() {
-      return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
-          .filter(FileStatus::isFile)
-          .map(stat -> {
-            InputFile file = HadoopInputFile.fromLocation(stat.getPath().toString(), conf);
-            long rowCount = Avro.rowCount(file);
-            Metrics metrics = new Metrics(rowCount, null, null, null, null);
-            String partitionKey = spec.fields().stream()
-                .map(PartitionField::name)
-                .map(name -> String.format("%s=%s", name, partitionPath.get(name)))
-                .collect(Collectors.joining("/"));
-
-            return DataFiles.builder(spec)
-                .withPath(stat.getPath().toString())
-                .withFormat("avro")
-                .withFileSizeInBytes(stat.getLen())
-                .withMetrics(metrics)
-                .withPartitionPath(partitionKey)
-                .build();
-
-          }).collect(Collectors.toList());
+      List<FileStatus> fileStatus = Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
+              .filter(FileStatus::isFile)
+              .collect(Collectors.toList());
+      DataFile[] datafiles = new DataFile[fileStatus.size()];
+      Tasks.Builder<Integer> task = Tasks.range(fileStatus.size())
+              .stopOnFailure()
+              .throwFailureWhenFinished();
+
+      if (parallelism > 1) {
+        task.executeWith(migrationService(parallelism));
+      }
+      
+      if (format.contains("avro")) {
+        task.run(index -> {
+          Metrics metrics = getAvroMerics(fileStatus.get(index), conf);
+          datafiles[index] = buildDataFile(fileStatus.get(index), partitionKey, spec, metrics, "avro");
+        });
+      } else if (format.contains("parquet")) {
+        task.run(index -> {
+          Metrics metrics = getParquetMerics(fileStatus.get(index), conf, metricsSpec, mapping);
+          datafiles[index] = buildDataFile(fileStatus.get(index), partitionKey, spec, metrics, "parquet");
+        });
+      } else if (format.contains("orc")) {
+        task.run(index -> {
+          Metrics metrics = getOrcMerics(fileStatus.get(index), conf, metricsSpec, mapping);
+          datafiles[index] = buildDataFile(fileStatus.get(index), partitionKey, spec, metrics, "avro");
+        });
+      } else {
+        throw new UnsupportedOperationException("Unknown partition format: " + format);
+      }
+      return Arrays.asList(datafiles);
     } catch (IOException e) {
       throw new RuntimeException("Unable to list files in partition: " + partitionUri, e);
     }
   }
 
-  private static List<DataFile> listParquetPartition(Map<String, String> partitionPath, String partitionUri,
-                                                     PartitionSpec spec, Configuration conf,
-                                                     MetricsConfig metricsSpec, NameMapping mapping) {
+  private static Metrics getAvroMerics(FileStatus stat,  Configuration conf) {
```

Comment:
One more minor thing I am noticing here, all these private methods pass through FileStatus, but they really only need  `String path`.

Reply:
Ok. I'm glad to do it .

Reply:
> > Looks weird, All these failed cases
> > ```shell
> >     Caused by: org.apache.avro.InvalidAvroMagicException: Not an Avro data file
> >     	at org.apache.avro.file.DataFileReader.openReader(DataFileReader.java:70)
> >     	at org.apache.iceberg.avro.AvroIterable.newFileReader(AvroIterable.java:100)
> > ```
> 
> Yeah something wrong with the format matching? Odd, for example the one I looked at is failing because an ORC file is being read as an AVRO one.


```shell
Error: eckstyle] [ERROR] /home/runner/work/iceberg/iceberg/data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java:91: Trailing whitespace is not allowed. [RegexpSinglelineJava]
Error: eckstyle] [ERROR] /home/runner/work/iceberg/iceberg/data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java:91: Whitespace at end-of-line [RegexpSingleline]
Error: eckstyle] [ERROR] /home/runner/work/iceberg/iceberg/data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java:105: Trailing whitespace is not allowed. [RegexpSinglelineJava]
> Task :iceberg-data:checkstyleMain FAILED
Error: eckstyle] [ERROR] /home/runner/work/iceberg/iceberg/data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java:105: Whitespace at end-of-line [RegexpSingleline]
```

this checkstyle error , can i just remove the empty line ?